include/lsst/sphgeom/CompoundRegion.h

```diff
@@ -95,11 +97,21 @@ class CompoundRegion : public Region {
     std::vector<std::uint8_t> _encode(std::uint8_t tc) const;
 
     // Implementation helper for decode().
-    static std::array<std::unique_ptr<Region>, 2> _decode(
+    static std::vector<std::unique_ptr<Region>> _decode(
         std::uint8_t tc, std::uint8_t const *buffer, std::size_t nBytes);
 
+    // Provide read access to all operands for quick iteration.
+    std::vector<std::unique_ptr<Region>> const& operands() const { return _operands; }
+
+    // Check if we have any operands.
+    bool empty() const { return _operands.empty(); }
```

Comment:
Given that `Region` also has an `isEmpty` method that's about whether there are any points in the region, I think this might be more confusing than helpful.

Reply:
OK, I'll drop it, I can use `nOperands()` which is public and will likely produce the same code.